src/sensai/evaluation/eval_stats/eval_stats_regression.py

```diff
@@ -0,0 +1,261 @@
+        pass
+
+    @classmethod
+    def computeErrors(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        return y_predicted - y_true
+
+    @classmethod
+    def computeAbsErrors(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        return np.abs(cls.computeErrors(y_true, y_predicted))
+
+
+class RegressionMetricMAE(RegressionMetric):
+    def __init__(self):
+        super().__init__("MAE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        return cls.computeAbsErrors(y_true, y_predicted)
+
+
+class RegressionMetricMSE(RegressionMetric):
+    def __init__(self):
+        super().__init__("MSE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        residuals = y_predicted - y_true
+        return np.sum(residuals * residuals) / len(residuals)
+
+
+class RegressionMetricRMSE(RegressionMetric):
+    def __init__(self):
+        super().__init__("MSE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        errors = cls.computeErrors(y_true, y_predicted)
+        return np.sqrt(np.mean(errors * errors))
+
+
+class RegressionMetricRRSE(RegressionMetric):
+    def __init__(self):
+        super().__init__("RRSE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        mean_y = np.mean(y_true)
+        residuals = y_predicted - y_true
+        mean_deviation = y_true - mean_y
+        return np.sqrt(np.sum(residuals * residuals) / np.sum(mean_deviation * mean_deviation))
+
+
+class RegressionMetricR2(RegressionMetric):
+    def __init__(self):
+        super().__init__("R2")
+
+    def computeValue(self, y_true: np.ndarray, y_predicted: np.ndarray):
+        rrse = RegressionMetricRRSE.computeValue(y_true, y_predicted)
+        return 1.0 - rrse*rrse
+
+
+class RegressionMetricPCC(RegressionMetric):
+    def __init__(self):
+        super().__init__("PCC")
+
+    def computeValue(self, y_true: np.ndarray, y_predicted: np.ndarray):
+        cov = np.cov([y_true, y_predicted])
+        return cov[0][1] / np.sqrt(cov[0][0] * cov[1][1])
+
+
+class RegressionMetricStdDevAE(RegressionMetric):
+    def __init__(self):
+        super().__init__("StdDevAE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        return np.std(cls.computeAbsErrors(y_true, y_predicted))
+
+
+class RegressionMetricMedianAE(RegressionMetric):
+    def __init__(self):
+        super().__init__("MedianAE")
+
+    @classmethod
+    def computeValue(cls, y_true: np.ndarray, y_predicted: np.ndarray):
+        return np.median(cls.computeAbsErrors(y_true, y_predicted))
+
+
+class RegressionEvalStats(PredictionEvalStats["RegressionMetric"]):
+    """
+    Collects data for the evaluation of predicted continuous values and computes corresponding metrics
+
+    :param y_predicted: the predicted values
```

Comment:
The params docstrings are in the wrong place.

Reply:
I did it on purpose. When the parameters are above the init, the rendered documentation looks much better (one can exclude the special member __init__ then). Otherwise I think part of the docu can get lost. At the same time pycharm is able to read the parameters from docstrings in this form. So I think there is no real downside to putting docstrings above the init. In some later commit I would adjust all remaining docstrings, if you agree

Reply:
I do not agree. The rendering of some piece of software should not have priority here.
The logical place to describe the constructor's arguments is in the constructor's dedicated docstring, while the class' docstring should describe what the class represents/does. Mixing that information with constructor-specific information is weird, especially since the class could provide alternative means of construction as well, and the constructor's args are just the "common denominator" (i.e. the most low-level data suitable for construction, typically) and not universally relevant. (I realise that calling the class is equivalent to calling __init__, so its not completely devoid of logic, but the argument still stands.)
Consistency with the rest of the project is also relevant; all other classes use the other style.
FWIW, PEP257 (Docstring Conventions) agrees with my notion: "The docstring for a class should summarize its behavior ... The class constructor should be documented in the docstring for its __init__ method. Individual methods should be documented by their own docstring."

Reply:
I'll change the docstring accordingly (directly on master)